### PR TITLE
[MOBI-2366] - inBrain SDK RN - add the new parameters

### PR DESCRIPTION
**Pull request by janosbrenda**

### CODE REVIEW

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 1]
```diff
@@ -230,20 +247,18 @@ public void nativeSurveysReceived(List<Survey> surveys) {
     // ***** SHOW NATIVE SURVEY ******
     // *******************************
     @ReactMethod
-    public void showNativeSurvey(final String id, final String searchId, final Promise promise) {
+    public void showNativeSurvey(final String id, final String searchId, final boolean showOffers, final Promise promise) {
```
Comment (olehhrechyn) [inline android/src/main/java/com/inbrain/InBrainSurveysModule.java:250]: I would like to propose renaming `showOffers` to `offersEnabled` to make all SDKs more similar, of course if it possible. At the moment the offers are showing, but who knows what the "enabled" flag will mean in the future.

**Review by olehhrechyn — CHANGES_REQUESTED** [review]
Great job @janosbrenda !
Looks good to me, I just left one small comment.

File: example/components/context/InbrainContext.tsx [hunk 2]
```diff
@@ -41,6 +42,7 @@ type InbrainContextType = {
   setSessionParameters: (sessionUid: string, dataPoints: DataPoints) => void;
   setOnCloseListener: (callback: () => void) => EmitterSubscription;
   setOnCloseListenerFromPage: (callback: () => void) => EmitterSubscription;
+  openWall: (option: InBrainWallOption) => Promise<void>;
```
Comment (susslik) [inline example/components/context/InbrainContext.tsx:45]: remove the `  showSurveys: () => Promise<void>;` and the related code;

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 3]
```diff
@@ -118,6 +118,23 @@ public void onFail(String message) {
     }
 
 
+
+    // ************************x
+    // ***** OPEN WALL ******
+    // ************************
+    @ReactMethod
+    public void openWall(final Promise promise) {
```
Comment (susslik) [inline android/src/main/java/com/inbrain/InBrainSurveysModule.java:126]: `option` parameter is missing

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 4]
```diff
@@ -118,6 +118,23 @@ public void onFail(String message) {
     }
 
 
+
```
Comment (susslik) [inline android/src/main/java/com/inbrain/InBrainSurveysModule.java:121]: Remove the `showSurveys(final Promise promise)` method

File: ios/InBrainSurveys.m [hunk 5]
```diff
@@ -138,12 +139,22 @@ + (BOOL)requiresMainQueueSetup {
     }];
 }
 
+// *******************************
```
Comment (susslik) [inline ios/InBrainSurveys.m:142]: remove the `showSurveys` method

File: src/InBrain.tsx [hunk 6]
```diff
@@ -86,10 +86,18 @@ const setOnSurveysCloseLister = (callback: (result: OnCloseSurveysData) => void
 const checkSurveysAvailable = () => wrapPromise<boolean>(() => InBrainSurveys.checkSurveysAvailable())
 
 /**
- * Show the Survey Wall. All the configs should be done `BEFORE` calling `showSurveys()`.
+ * @deprecated Use openWall() instead
  */
 const showSurveys = () => wrapPromise<void>(() => InBrainSurveys.showSurveys());
```
Comment (susslik) [inline src/InBrain.tsx:91]: 1) call the new `openWall` method inside the `showSurveys`, i.e.
`const showSurveys = () => wrapPromise<void>(() => openWall(InBrainWallOption.all));`

2) move the method bellow the actual methods, i.e. add the new //--- Deprecated -- section above the `Unsupported`

**Review by susslik — CHANGES_REQUESTED** [review]

**Review by susslik — COMMENTED** [review]
I went ahead and fixed all the issues and missing things, so assume it works now. 
However needs the review from Kiro/Oleh to verify we haven't miss anything.
Also before the release - pls, ensure the version of the SDK is increased, the readme is updated and the versions to the native dependencies set to correct values

**Review by olehhrechyn — APPROVED** [review]

File: android/build.gradle [hunk 7]
```diff
@@ -71,5 +71,5 @@ repositories {
 dependencies {
     //noinspection GradleDynamicVersion
     implementation 'com.facebook.react:react-native:+'
-    implementation 'com.github.inbrainai:sdk-android:[2.4.5, 3.0.0)'
+    implementation 'com.github.inbrainai:sdk-android:a400a045438564d0e980826815abd3e554836243'
```
Comment (kiroj305) [inline android/build.gradle:74]: Please use latest version 2.5.0 now.

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 8]
```diff
@@ -92,12 +93,29 @@ public void setDataOptions(final ReadableMap data) {
...
     }
 
+
+    // ************************x
+    // ***** OPEN WALL ******
     // ************************
-    // ***** SHOW SURVEYS *****
-    // ************************
+
+    // TODO: - Move the initFromRaw to Android SDK
```
Comment (kiroj305) [inline android/src/main/java/com/inbrain/InBrainSurveysModule.java:101]: You can do this `TODO` right now with 2.5.0.

**Review by kiroj305 — CHANGES_REQUESTED** [review]

File: example/ios/Podfile [hunk 9]
```diff
@@ -17,6 +17,8 @@ end
 target 'InbrainReactNativeExample' do
   config = use_native_modules!
 
+  pod 'InBrainSurveys', :git => 'https://github.com/inbrainai/inBrainSurveys_SDK.git', :branch => 'develop'
```
Comment (susslik) [inline example/ios/Podfile:20]: Also needs to point to the correct version

**Review by susslik — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 10]
```diff
@@ -94,46 +94,35 @@ public void setDataOptions(final ReadableMap data) {
...
-        return WallOption.ALL;
-    }
-
+// ************************
+// ***** OPEN WALL ******
+// ************************
     @ReactMethod
     public void openWall(final int option, final Promise promise) {
-        WallOption wallOption = mapRawToOption(option);
+    WallOption wallOption = WallOption.fromRaw(option);
```
Comment (susslik) [inline android/src/main/java/com/inbrain/InBrainSurveysModule.java:102]: That does not work. Pls, check the comment for Flutter SDK.
And pls, run the app before asking about the review

File: example/ios/Podfile [hunk 11]
```diff
@@ -17,6 +17,8 @@ end
 target 'InbrainReactNativeExample' do
   config = use_native_modules!
 
+  pod 'InBrainSurveys', :git => 'https://github.com/inbrainai/inBrainSurveys_SDK.git', :tag => '2.5.0'
```
Comment (susslik) [inline example/ios/Podfile:20]: Hmmm... that shouldn't be added. It was just for testing purposes during development

File: inbrain-surveys.podspec [hunk 12]
```diff
@@ -20,6 +20,6 @@ Pod::Spec.new do |s|
   s.source_files = "ios/*.{h,m,swift}"
 
   s.dependency "React"
-  s.dependency 'InBrainSurveys', '~> 2.4.3'
+  s.dependency 'InBrainSurveys'
```
Comment (susslik) [inline inbrain-surveys.podspec:23]: the version range is missing, should be:
 s.dependency 'InBrainSurveys', '~> 2.5.0'

**Review by susslik — CHANGES_REQUESTED** [review]

File: example/ios/Podfile (hunk 9 above)
Comment (susslik) [reply example/ios/Podfile]: done

File: example/ios/Podfile (hunk 11 above)
Comment (susslik) [reply example/ios/Podfile:20]: Done

File: inbrain-surveys.podspec (hunk 12 above)
Comment (susslik) [reply inbrain-surveys.podspec:23]: Done

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java (hunk 10 above)
Comment (susslik) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java]: Done

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java (hunk 8 above)
Comment (susslik) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java]: done

File: android/build.gradle (hunk 7 above)
Comment (susslik) [reply android/build.gradle]: done

**Review by susslik — APPROVED** [review]
Looks good to me! @olehhrechyn , @kiroj305 could you take a look pls as wel to verify I haven't missed anything?

**Review by kiroj305 — APPROVED** [review]
Great work on bunch of refactoring things! I haven't found anything missing.

**Review by olehhrechyn — CHANGES_REQUESTED** [review]
The same change as in Flutter,
`implementation 'com.github.inbrainai:sdk-android:[2.5.0, 3.0.0)'`

**Review by olehhrechyn — APPROVED** [review]

**Pull request merged**